### PR TITLE
Renamed `Natural-key slug` to `Composite key` throughout the UI and documentation.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #3793 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/models/constants.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+# For our purposes, COMPOSITE_KEY_SEPARATOR needs to be:
+# 1. Safe in a URL path component (so that we can do URLS like "/dcim/devices/<composite_key>/delete/")
 #    Per RFC3986 section 2.3 the general "unreserved" characters are ALPHA and DIGIT and the characters -._~
 #    Per RFC3986 section 3.3 path components also permit characters :@ and the "sub-delims" characters !$&'()*+,;=
 # 2. Not readily confused with characters commonly seen in un-escaped natural key component fields
 #    "." is already ruled out as an unreserved character but also would appear in IPv4 IPAddress and Prefix objects
 #    ":" similarly would appear in IPv6 IPAddress/Prefix objects
 #    "/" would appear in Prefix objects as well as various numbered device component names
-NATURAL_KEY_SLUG_SEPARATOR = "&"
+COMPOSITE_KEY_SEPARATOR = "&"
```
Comment (glennmatthews) [inline nautobot/core/models/constants.py:11]: Not blocking but let's think about revisiting this in the very near future - I'm just thinking that `&` is going to be problematic in query strings if/when we want NaturalKeyOrPKMultipleChoiceFilter to support composite-keys (`/dcim/devices/?location=Campus&Location1` will not work). Maybe we should change this to `!` or `;`? Needs some thought.

File: nautobot/core/models/querysets.py [hunk 2]
```diff
@@ -102,14 +102,14 @@ def distinct_values_list(self, *fields, flat=False, named=False):
 
     def filter(self, *args, **kwargs):
         """
-        Extend base queryset with support for filtering by `natural_key_slug=...`.
+        Extend base queryset with support for filtering by `composite_key=...`.
 
-        This is an enhanced version of natural-key slug support from django-natural-keys.
-        Counterpart to BaseModel.natural_key_slug property.
+        This is an enhanced version of composite key support from django-natural-keys.
```
Comment (glennmatthews) [inline nautobot/core/models/querysets.py:107]:
```suggestion
        This is an enhanced version of natural-key slug key support from django-natural-keys.
```

Since django-natural-keys uses "natural-key slug" instead of "composite key" terminology.

File: nautobot/core/models/constants.py (hunk 1 above)
Comment (HanlinMiao) [reply nautobot/core/models/constants.py]: Changing it to `;` per discussion in standup.

File: nautobot/core/models/constants.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-# For our purposes, NATURAL_KEY_SLUG_SEPARATOR needs to be:
-# 1. Safe in a URL path component (so that we can do URLS like "/dcim/devices/<natural_key_slug>/delete/")
+# For our purposes, COMPOSITE_KEY_SEPARATOR needs to be:
+# 1. Safe in a URL path component (so that we can do URLS like "/dcim/devices/<composite_key>/delete/")
 #    Per RFC3986 section 2.3 the general "unreserved" characters are ALPHA and DIGIT and the characters -._~
 #    Per RFC3986 section 3.3 path components also permit characters :@ and the "sub-delims" characters !$&'()*+,;=
 # 2. Not readily confused with characters commonly seen in un-escaped natural key component fields
 #    "." is already ruled out as an unreserved character but also would appear in IPv4 IPAddress and Prefix objects
 #    ":" similarly would appear in IPv6 IPAddress/Prefix objects
 #    "/" would appear in Prefix objects as well as various numbered device component names
```
Comment (glennmatthews) [inline nautobot/core/models/constants.py:10]: Perhaps add a comment like:

```
# 3. Safe in a URL query string component (so that we can do URLs like "/dcim/devices/?location=<composite_key>"
#    This rules out "&" and "="
```

File: changes/3793.changed [hunk 4]
```diff
@@ -0,0 +1 @@
+Renamed Natural-key slug to Composite key throughout the UI and documentation.
```
Comment (glennmatthews) [inline changes/3793.changed:1]: Please add a note about the change in separator character as well.

File: nautobot/docs/release-notes/version-2.0.md [hunk 5]
```diff
@@ -316,7 +316,7 @@ Support for RQ and `django-rq`, deprecated since Nautobot 1.1.0, has been fully
 - [#3715](https://github.com/nautobot/nautobot/issues/3715) - Added (temporary) `natural_key_field_names` to `IPAddress`, `Prefix`, `RackReservation`, `ScheduledJob`, and `Service` models.
 - [#3721](https://github.com/nautobot/nautobot/issues/3721) - Added App provided model view override.
 - [#3722](https://github.com/nautobot/nautobot/issues/3722) - Added `termination_type` filter to `CableFilterSet`.
-- [#3722](https://github.com/nautobot/nautobot/issues/3722) - Added `natural_key_slug` field to REST API serializers.
+- [#3722](https://github.com/nautobot/nautobot/issues/3722) - Added `composite_key` field to REST API serializers.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:319]: I'd say revert this change - we shouldn't go back and change old release-notes.

File: nautobot/docs/release-notes/version-2.0.md [hunk 6]
```diff
@@ -33,10 +33,10 @@ Nautobot's `BaseModel` base class and related classes now implement automatic su
 >>> DeviceType.objects.get_by_natural_key("MegaCorp", "Model 9000")
 <DeviceType: Model 9000>
 
->>> DeviceType.objects.first().natural_key_slug
+>>> DeviceType.objects.first().composite_key
 'MegaCorp&Model+9000'
 
->>> DeviceType.objects.get(natural_key_slug="MegaCorp&Model+9000")
+>>> DeviceType.objects.get(composite_key="MegaCorp&Model+9000")
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:39]:
```suggestion
>>> DeviceType.objects.get(composite_key="MegaCorp;Model+9000")
```

**Review by glennmatthews — APPROVED** [review]
